helpers/TestCenterHelper.php

```diff
@@ -239,153 +238,80 @@ public static function removeDiagnostic(core_kernel_classes_Resource $testCenter
-                'actor' => 'Billy LaPorte',
-                'event' => 'start item',
-                'details' => '',
-                'context' => 'Section 1 - item 2/3'
-            ];
-            $history[] = [
-                'timestamp' => $startTime + rand(2010, 2020),
-                'session' => $sessionUri,
-                'role' => 'Test Taker',
-                'actor' => 'Billy LaPorte',
-                'event' => 'prohibited',
-                'details' => 'shortcut: Cmd+c',
-                'context' => 'Section 1 - item 2/3'
-            ];
-            $history[] = [
-                'timestamp' => $startTime + rand(2020, 2030),
-                'session' => $sessionUri,
-                'role' => 'Test Taker',
-                'actor' => 'Billy LaPorte',
-                'event' => 'end item',
-                'details' => '',
-                'context' => 'Section 1 - item 2/3'
-            ];
-            $history[] = [
-                'timestamp' => $startTime + rand(2010, 2020),
-                'session' => $sessionUri,
-                'role' => 'Test Taker',
-                'actor' => 'Billy LaPorte',
-                'event' => 'start item',
-                'details' => '',
-                'context' => 'Section 1 - item 3/3'
-            ];
-            $history[] = [
-                'timestamp' => $startTime + rand(2040, 2050),
-                'session' => $sessionUri,
-                'role' => 'Proctor',
-                'actor' => 'Proctor 1',
-                'event' => 'terminate',
-                'details' => ['Fire emergency', 'Exit code: T'],
-                'context' => 'Section 1 - item 3/3'
-            ];
+            $valid = false;
+            $deliveryExecution = \taoDelivery_models_classes_execution_ServiceProxy::singleton()->getDeliveryExecution($sessionUri);
+            $delivery = $deliveryExecution->getDelivery();
+            $executions = $deliveryService->getCurrentDeliveryExecutions($delivery, $testCenter, $options);
+            foreach($executions as $execution){
+                if($execution['delivery_execution_id'] === $sessionUri && $execution['authorized_by'] === $proctor->getUri()){
+                    $valid = true;
+                    continue;
+                }
+            }
+            if(!$valid){
+                continue;
+            }
+
+            if($logHistory){
+                $deliveryLog->log($deliveryExecution->getIdentifier(), 'HISTORY', array());
+            }
+
+            $logs = $deliveryLog->get($deliveryExecution->getIdentifier());
+            $exportable = array();
+            foreach($logs as $data){
+                if($data['event_id'] !== 'HEARTBEAT' && strpos($data['event_id'], 'latency') === FALSE){
+                    $author = new \core_kernel_classes_Resource($data['created_by']);
+                    $proctorRole = new \core_kernel_classes_Resource('http://www.tao.lu/Ontologies/TAOProctor.rdf#ProctorRole');
```

Comment:
I don't known what is the performances impact of creating an instance of Resource, but maybe it could be better to factorise this variable and move this line outside of the loops...